sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -145,3 +146,23 @@ def test_should_stop(scores, n_iter_no_change, tol, stopping):
         n_iter_no_change=n_iter_no_change, tol=tol
     )
     assert gbdt._should_stop(scores) == stopping
+
+
+def test_binning_train_validation_are_separated():
```

Comment:
Since we know `n_samples` and `validation_fraction`, we can test the `actual_n_bins` directly:

```py
def test_binning_train_validation_are_separated():
    # Make sure training and validation data are binned separately.
    # See issue 13926

    rng = np.random.RandomState(0)
    n_samples = X_classification.shape[0]
    valid_frac = 0.2
    gb = HistGradientBoostingClassifier(n_iter_no_change=5,
                                        validation_fraction=valid_frac,
                                        random_state=rng)
    gb.fit(X_classification, y_classification)
    mapper_training_data = gb.bin_mapper_

    assert np.all(mapper_training_data.actual_n_bins_ ==
                  int((1 - valid_frac) * n_samples))
```

Reply:
I'd rather explicitly have a comparison operator here, in order to reflect the intent of the test

Reply:
I used both